integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/reference/FieldReferenceIT.java

```diff
@@ -0,0 +1,524 @@
+						LocalDate.class );
+				vector = ValueFieldReference.of( absolutePath + ".vector", float[].class, float[].class, float[].class );
+				stringList = ValueFieldReference.of( absolutePath + ".stringList", String.class, String.class, String.class );
+
+				flattenedEmbeddedEntity = new FlattenedEmbeddedEntity_( absolutePath + ".flattenedEmbeddedEntity" );
+				nestedEmbeddedEntity = new NestedEmbeddedEntity_( absolutePath + ".nestedEmbeddedEntity" );
+				embeddedEntityList = new EmbeddedEntityList_( absolutePath + ".embeddedEntityList" );
+
+				parent = new Parent__( absolutePath + ".parent" );
+			}
+
+			public static class NestedEmbeddedEntity_ extends ObjectFieldReferenceImpl implements NestedObjectFieldReference {
+
+				public ValueFieldReference<String, String, String> string;
+				public ValueFieldReference<SomeEnum, String, SomeEnum> someEnum;
+
+				private NestedEmbeddedEntity_(String absolutePath) {
+					super( absolutePath );
+					this.string = ValueFieldReference.of( absolutePath + ".string", String.class, String.class, String.class );
+					this.someEnum =
+							ValueFieldReference.of( absolutePath + ".someEnum", SomeEnum.class, String.class, SomeEnum.class );
+				}
+			}
+
+			public static class EmbeddedEntityList_ extends ObjectFieldReferenceImpl implements NestedObjectFieldReference {
+
+				public ValueFieldReference<String, String, String> string;
+				public ValueFieldReference<SomeEnum, String, SomeEnum> someEnum;
+
+				private EmbeddedEntityList_(String absolutePath) {
+					super( absolutePath );
+					this.string = ValueFieldReference.of( absolutePath + ".string", String.class, String.class, String.class );
+					this.someEnum =
+							ValueFieldReference.of( absolutePath + ".someEnum", SomeEnum.class, String.class, SomeEnum.class );
+				}
+			}
+
+			public static class FlattenedEmbeddedEntity_ extends ObjectFieldReferenceImpl {
+
+				public ValueFieldReference<String, String, String> string;
+				public ValueFieldReference<SomeEnum, String, SomeEnum> someEnum;
+
+				private FlattenedEmbeddedEntity_(String absolutePath) {
+					super( absolutePath );
+					this.string = ValueFieldReference.of( absolutePath + ".string", String.class, String.class, String.class );
+					this.someEnum =
+							ValueFieldReference.of( absolutePath + ".someEnum", SomeEnum.class, String.class, SomeEnum.class );
+				}
+			}
+
+			public static class Parent__ extends ObjectFieldReferenceImpl implements NestedObjectFieldReference {
+
+				public ValueFieldReference<String, String, String> string;
+				public ValueFieldReference<float[], float[], float[]> vector;
+
+				private Parent__(String absolutePath) {
+					super( absolutePath );
+					this.string = ValueFieldReference.of( absolutePath + ".string", String.class, String.class, String.class );
+					this.vector =
+							ValueFieldReference.of( absolutePath + ".vector", float[].class, float[].class, float[].class );
+				}
+			}
```

Comment:
For example, this `parent` only has two fields since others are excluded. And we stop here because of the specified depth.